File: frontend/src/models/shot.ts
export type Shot = {
	shot_id: string
	brew_time: string // ISO string from backend
	machine_id: string
	user_id: string
	software_bundle: string
	coffee_type: string
	recipe_id: string
	grind_size_actual: number
	grind_size_target: number
	dose_grams: number
	dose_target_grams: number
	brew_time_seconds: number
	peak_pressure_bar: number
	last_status: string
}

File: backend/internal/domain/shot.go
package domain

import "time"

// Shot models one espresso shot record in the core domain.
type Shot struct {
	ShotID          string
	BrewTime        time.Time
	MachineID       string
	UserID          string
	SoftwareBundle  string
	CoffeeType      string
	RecipeID        string
	GrindSizeActual int
	GrindSizeTarget int
	DoseGrams       float64
	DoseTargetGrams float64
	BrewTimeSeconds float64
	PeakPressureBar float64
	LastStatus      string
}
